### ISSUE TITLE
Organise grapqhl queries and mutations

### ISSUE THREAD

**Issue by vamshi9666** [coderplex/learnplex#142]
**Is your feature request related to a problem? Please describe.**

Currently all pages and components which needs a query or mutation is using a query declared in that component itself.
If though two or more components needs exact data. same query is declared in two places. This voilates dry code rule and even need to check both places if any typo or some error occurs

**Describe the solution you'd like**
It would be better to have all queries , mutations , subs in one place.

--queries\
   |--user.js
   |--resources.js
--mutations\
  |--user.js
  |--topic.js

**Comment by vinaypuppal** [comment]
@pbteja1998 I wonder why you are not using GraphQL codegen for `urql`?

Did you check this https://graphql-code-generator.com/docs/plugins/typescript-urql

This generates hooks from GraphQL queries/mutations on the client-side. This solves this issue so all queries and mutations stay in `graph.gql` files and hooks will be generated by the above plugin which we can import in any component

**Comment by pbteja1998** [comment]
@vamshi9666 Yeah, I am organising the code in a similar manner. Right now, I am moving out the code related to queries and mutations to `/utils`. I have already moved out most of the queries and mutations. Will eventually organise them in a way that you proposed. 

@vinaypuppal I have previously used codegen and generated hooks when I was using apollo instead of urql. I faced some problems with it. I was unable to use them in some places like inside useEffect and in some other functions(since they are hooks). Moreover I thought that even though I was using same query at different places, the fields that I was selecting was different in different cases.So having autogenerated hooks from centralized .gql files did not help. 

But I think we can use a mix of autogenerated hooks and directly calling `urqlClient.query()` at places where hooks cannot be used. Will give codegen gql hooks another try.

**Comment by vamshi9666** [comment]
> But I think we can use a mix of autogenerated hooks and directly calling `urqlClient.query()` at places where hooks cannot be used. Will give codegen gql hooks another try.

Have you tried using useLazyQuery in that case ?

**Comment by pbteja1998** [comment]
> > But I think we can use a mix of autogenerated hooks and directly calling `urqlClient.query()` at places where hooks cannot be used. Will give codegen gql hooks another try.
> 
> Have you tried using useLazyQuery in that case ?

This is still a hook, right? I am talking about a case where hooks cannot be used, like inside a function in /api.

**Comment by vamshi9666** [comment]
useLazyQuery will only return a function right? it can be used any where you need. even inside useeffect or even general function. 
```jsx
const [fetchUsers, usersData] = useLazyQuery()


const someFunction = () => {
  restapi.getUsers().then( () => {
    fetchUsers()
  })

 return (...jsx)
}
````

**Comment by vinaypuppal** [comment]
Urql does not support lazy queries. Instead it supports pausing https://formidable.com/open-source/urql/docs/basics/queries/#pausing-usequery
